File: golang/1754. 构造字典序最大的合并字符串.go
func largestMerge(word1 string, word2 string) string {
    merge:=[]byte{}
    i,j,n,m:=0,0,len(word1),len(word2)
    for i<n||j<m {
        if i<n &&word1[i:]>word2[j:]{
            merge=append(merge,word1[i])
            i+=1
        }else{
            merge=append(merge,word2[j])
            j+=1
        }
    }
    return string(merge)
}

File: 刷题/cpp/1754. 构造字典序最大的合并字符串.cpp
class Solution {
public:
    string largestMerge(string word1, string word2) {
        string merge="";
        while(word1.length()&&word2.length()){
            if(word1>word2){
                merge+=word1[0];
                word1.erase(word1.begin());
            }else{
                merge+=word2[0];
                word2.erase(word2.begin());
            }
        }
        if(word1.length()){
            merge+=word1;
        }
        if(word2.length()){
            merge+=word2;
        }
        return merge;
    }
};